controllers/apps/transformer_rbac.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package apps
+
+import (
+	"github.com/spf13/viper"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/apecloud/kubeblocks/internal/controller/builder"
+	"github.com/apecloud/kubeblocks/internal/controller/graph"
+	ictrltypes "github.com/apecloud/kubeblocks/internal/controller/types"
+)
+
+// RBACTransformer puts the rbac at the beginning of the DAG
+type RBACTransformer struct{}
+
+var _ graph.Transformer = &RBACTransformer{}
+
+func (c *RBACTransformer) Transform(ctx graph.TransformContext, dag *graph.DAG) error {
+	transCtx, _ := ctx.(*ClusterTransformContext)
+	cluster := transCtx.Cluster
+	if !viper.GetBool("ENABLE_RBAC_MANAGER") {
+		transCtx.Logger.Info("rbac manager is not enabled")
+		return nil
+	}
+
+	root, err := ictrltypes.FindRootVertex(dag)
+	if err != nil {
+		return err
+	}
+
+	for _, compSpec := range cluster.Spec.ComponentSpecs {
+		serviceAccountName := compSpec.ServiceAccountName
+		if isServiceAccountNotExist(transCtx, serviceAccountName) {
+			serviceAccount, err := builder.BuildServiceAccount(cluster)
+			if err != nil {
+				return err
+			}
+			serviceAccount.Name = serviceAccountName
```

Comment:
SA and RoleBinding are cluster scope resources, it's weird to handle them inside a component

Reply:
clusterrole are cluster scoped, SA and RoleBinding are namespace scoped, and every cluster has one of them